### PR TITLE
[fix](DECIMAL) error DECIMAL cat to BOOLEAN

### PR DESCRIPTION
**Pull request by Mryange**
### What problem does this PR solve?

In the past, there were issues with converting `double` and `decimal` to `boolean`.
For example, a `double` value like 0.13 would first be cast to `uint8`, resulting in 0.
Then, it would be converted to `bool`, yielding 0 (incorrect, as the expected result is 1).

Similarly, `decimal` values were directly cast to `uint8`, leading to non-0/1 values for `bool`.
This issue arises because Doris internally uses `uint8` to represent `boolean`.

before
```
mysql> select cast(40.123 as BOOLEAN);
+-------------------------+
| cast(40.123 as BOOLEAN) |
+-------------------------+
|                      40 |
+-------------------------+
```

now
```
mysql> select cast(40.123 as BOOLEAN);
+-------------------------+
| cast(40.123 as BOOLEAN) |
+-------------------------+
|                       1 |
+-------------------------+
```

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [x] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by Mryange** [comment]
run external

**Review by HappenLee — APPROVED** [review]
LGTM

File: regression-test/suites/correctness/test_cast_decimalv3_as_bool.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            `k2` decimalv3(18,9) ,
+            `k3` decimalv3(38,16) ,
+        )
+        UNIQUE KEY(`id`)
+        DISTRIBUTED BY HASH(`id`) BUCKETS 10
+        PROPERTIES (
+        "enable_unique_key_merge_on_write" = "true",
+        "replication_num" = "1"
+        );
+    """
```
Comment (yiguolei) [inline regression-test/suites/correctness/test_cast_decimalv3_as_bool.groovy:33]: 1. 测试一下常量： select 1.1 as boolean
2. 测试一下所有的标量类型： 数值：tinyint，float，double，int，short ； 日期date，datetime.

11100000000  uint8

**Pull request merged**